Create Quiz For Friends

louisavasies opened the pull request.

louisavasies committed:
Start Game Page

andrianwritek commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,44 @@
+                        </div>
+                    </div>
+                </div>
+                <footer id="footer"><BottomLogo /></footer>  
```
why footer is not in master view

andrianwritek commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,44 @@
+
+
+    public render() {
+        const randomstring = require("randomstring");
```
Failed to compile.

./src/Components/StartGame/StartGame.tsx
Module not found: Can't resolve 'randomstring' in 'C:\repo\VikinQuiz\VikingQuiz.UI\src\Components\StartGame'

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.css:
```diff
@@ -0,0 +1,59 @@
+.codeLabel{
```
Please use kebab-case for class names in CSS

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.css:
```diff
@@ -0,0 +1,59 @@
+    text-align: center;
+}
+
+.homeButton{
```
Please use kebab-case for class names in CSS

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.css:
```diff
@@ -0,0 +1,59 @@
+
+
+@media screen and (max-height: 800px){
+    .homeButton{
```
Please use kebab-case for class names in CSS

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.css:
```diff
@@ -0,0 +1,59 @@
+        height:240px;
+    }
+    
+    .codeLabel{
```
Please use kebab-case for class names in CSS

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,44 @@
+                    </div>
+                    <div className="row">
+                        <div className="col-xs-10 col-xs-offset-1 col-md-6 col-md-offset-3">
+                            <div className="codeLabel"> YOUR CODE </div>
```
It would be cool if we can extract this string "YOUR CODE" to some constant, so that we can easy change it in future (or maybe in case we want to add localization).

louisavasies committed:
Added kebab-case for all CSS; constant message in StartGame page.

vasi-axinte commented on VikingQuiz.UI/src/Components/FormComponent/FormInput/FormInput.tsx:
```diff
@@ -3,10 +3,10 @@ import './FormInput.css';
-        <div className="formerror">{props.ErrorMessage}</div>
+       <div className="form-group col-xs-12">
+        <label className="form-label" htmlFor={props.InputId}>{props.InputLabel}</label>
+        <input id={props.InputId} name={props.name} className="my-input form-control" type={props.InputType} onChange={props.changed}/>
```
The class my-input doesn`t look good. Please rename it to something more meaningful

andrianwritek commented:
Failed to compile
./src/Components/StartGame/StartGame.tsx
Module not found: Can't resolve 'randomstring' in 'C:\repo\VikinQuiz\VikingQuiz.UI\src\Components\StartGame'
This error occurred during the build time and cannot be dismissed.

andrianwritek reviewed (dismissed):
change footer, all rest is cool

louisavasies committed:
Added Play Button, resolved PlayerController for GET; Almost Finished StartGame Page

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerController.cs:
```diff
@@ -27,7 +27,7 @@ public PlayerController(PlayerRepo playerRepo, IEntityMapper<Player, PlayerViewM
         public IActionResult GetPlayer()
         {
-            var result = playerRepo.GetAllPlayers().Select(s => entityToVmMapper.Map(s)).ToList();
+            var result = playerRepo.GetAll().Select(s => entityToVmMapper.Map(s)).ToList();
```
Please use a more meaningful letter in lambda expressions. For example, if you want to select players, use "p" instead of "s".

vasi-axinte commented on VikingQuiz.Api/Repositories/GameRepo.cs:
```diff
@@ -27,6 +27,7 @@ public Game Update(Game g)
             Game gg = ctx.Game.Find(g.Id);
```
Where does the gg abreviation comes from? I think a more meaningful name should be used. Maybe "game" is a better idea?

Commit:
- added Migration for project
- added new field "Code" in Game table
- reverted "hack" for migrations

louisavasies committed:
Made page responsive; Added Start and Cancel Game Buttons

vasi-axinte commented on VikingQuiz.Api/Controllers/GameController.cs:
```diff
@@ -47,10 +48,12 @@ public IActionResult GetAnswerById(int id)
         public IActionResult CreateGame([FromBody]GameViewModel game)
         {
+            string code = RandomCode(6);
             Game g = new Game()
```
This should be changed to "game" instead of g.

vasi-axinte commented on VikingQuiz.UI/src/Components/FormComponent/FormInput/FormInput.css:
```diff
@@ -15,7 +15,7 @@
 
-.formerror{
+.form-error{
     /* position: absolute; */ /*important! otherwise element will take up space when shown
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    public componentWillMount() {
+        axios.get('http:///localhost:60151/api/player')
```
Please extract the url to a constant so it would be easier to be changed.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,95 @@
+
+    public getFirstListOfPlayers() {
+        const firstListOfPlayers = [];
+        for( let i = 0; i < 7; i++){
```
This magic number 7 should be extracted in a constant in case in the future somebody would like it to become 8 or 10 or 5.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -0,0 +1,95 @@
+        let firstListOfPlayers = [];
+        let secondListOfPlayers = [];
+        if (this.state.player.length > 7){
+            firstListOfPlayers = this.getFirstListOfPlayers();
```
This is wrong. You should rethink the algorithm so that you don`t need to use an if anymore.

vasi-axinte commented on VikingQuiz.UI/src/Components/UserMinimalProfile/UserMinimalProfile.tsx:
```diff
@@ -0,0 +1,28 @@
+        return (
+            <div className="player-container"> 
+                    <div className="player-photo">
+                        <img className="photo" src="http://mythologian.net/wp-content/uploads/2017/02/real-floki-from-vikings-1024x657.jpg"/>                      
```
Why is this picture hardcoded?

louisavasies committed:
Changed algorithm for showing players; Deleted comments

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerRepo.cs:
```diff
@@ -38,9 +38,17 @@ public Player UpdatePlayer(Player player)
             return ctx.Player.ToList();
         }
+
+        //public void connectToGame(Player player, Game game, string code)
```
Please remove this commented code.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/StartGame.tsx:
```diff
@@ -14,48 +14,23 @@ class StartGame extends React.Component<any, any> {
 
     public componentWillMount() {
-        axios.get('http:///localhost:60151/api/player')
+        axios.get(this.state.urlToGet)
```
urlToGet is not a good name. You should build the url in the following way: base + endpoint where base = http://localhost:60151/api and endpoint will be "player". That way, you will reuse a lot of that code.

vasi-axinte commented:
Please do not forget to fix the conflicts before merge

louisavasies committed:
Finished pages for creating a player for a game in progress and connectig the player to it;
Added PlayerGameViewModel and PlayerGameController+Repo + Mappers;
Added DTO for player+code;
Finished PlayerController for having codes;
Changed GameController with updated version;

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerGameController.cs:
```diff
@@ -0,0 +1,80 @@
+    [Route("api/[controller]")]
+    public class PlayerGameController : Controller
+    {
+        private readonly PlayerGameRepo playergameRepo;
```
Please do not abbreviate the repository word

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerGameController.cs:
```diff
@@ -0,0 +1,80 @@
+        [HttpGet]
+        public IActionResult GetPlayerGame()
+        {
+            var result = playergameRepo.GetAll().Select(s => entityToVmMapper.Map(s)).ToList();
```
Why do you use s as the lambda function parameter? What does it stand for?

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerGameController.cs:
```diff
@@ -0,0 +1,80 @@
+        {
+            PlayerGame playergame = new PlayerGame()
+            {
+                Pid = playergameViewModel.Pid, 
```
What is the meaning of Pid? What about Gid?

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerGameController.cs:
```diff
@@ -0,0 +1,80 @@
+        [HttpPut("{id}")]
+        public IActionResult UpdatePlayerGame(int pid, int gid, [FromBody]PlayerGameViewModel playergame)
+        {
+            PlayerGame pg = new PlayerGame()
```
Please use full name for a variable.

vasi-axinte commented on VikingQuiz.Api/Repositories/GameRepo.cs:
```diff
@@ -9,6 +9,7 @@ namespace VikingQuiz.Api.Repositories
     public class GameRepo
     {
         private VikinQuizContext ctx;
```
Please do not abbreviate the context to ctx

vasi-axinte commented on VikingQuiz.Api/Repositories/GameRepo.cs:
```diff
@@ -53,6 +54,27 @@ public Game GetGameById(int id)
+        public Game GetGameByCode(string code)
+        {
+            Game foundGame = ctx.Game.Where(x => x.Code == code)
+                .Select(x => new Game { Id = x.Id, QuizId = x.QuizId, GameDate = x.GameDate, Code = x.Code })
```
A mapping method would have looked more nice here instead of creating the Game inside the select. Also, please use meaningful names for lambda function parameters

vasi-axinte commented on VikingQuiz.Api/Repositories/GameRepo.cs:
```diff
@@ -53,6 +54,27 @@ public Game GetGameById(int id)
+        {
+            const string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+            string code = "";
+            for(int i = 0; i < 6; i++)
```
So many magic numbers here in this method. We should use the length of chars string instead of 35 and some constant instead of the number 6

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerGameRepo.cs:
```diff
@@ -0,0 +1,52 @@
+{
+    public class PlayerGameRepo
+    {
+        private VikinQuizContext ctx;
```
Same story with the ctx variable.

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerGameRepo.cs:
```diff
@@ -0,0 +1,52 @@
+
+        public void Delete(int pid, int gid)
+        {
+            PlayerGame pg = new PlayerGame { Pid = pid, Gid = gid };
```
Abbreviations, abbreviations...

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerRepo.cs:
```diff
@@ -43,12 +48,36 @@ public List<Player> GetAll()
-        //{
-        //    if (code == game.Code)
-        //    {
+        public Player getPlayerById(int id)
```
Why is this method name camelCased?

vasi-axinte commented on VikingQuiz.Api/ViewModels/PlayerGameViewModel.cs:
```diff
@@ -0,0 +1,14 @@
+{
+    public class PlayerGameViewModel
+    {
+        public int Pid { get; set; }
```
Please change these to PlayerId and GameId or whatever their meaning is.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/ConnectGame.tsx:
```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import './ConnectGame.css';
+// import axios from '../../../node_modules/axios';
```
Please remove unused imports.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/ConnectGame.tsx:
```diff
@@ -0,0 +1,56 @@
+            serverMessage: '',
+            redirect: false,
+            player: [],
+            // baseUrl : "http:///localhost:60151/api/",
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/ConnectGame.tsx:
```diff
@@ -0,0 +1,56 @@
+                            <div className="col-sm-auto">
+                                <label className="connectgame-main-label">{mainMessage}</label>
+                            </div>
+                            {/* <div className="connectgame-players-container">
```
Please remove commented code

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGame.tsx:
```diff
@@ -0,0 +1,87 @@
+    
+        const body: PlayerCodeDto = new PlayerCodeDto('', formData.PlayerName, formData.GameCode);
+    
+        console.log(body);
```
Please remove console logs from your code.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGame.tsx:
```diff
@@ -0,0 +1,87 @@
+                return;
+            };
+            comp.setState({
+                serverMessage: error.response.data
```
What does comp stands for?

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGame.tsx:
```diff
@@ -0,0 +1,87 @@
+                                        new InputData('code', 'text', 'Enter your code/pin', '', 'GameCode', ''),
+                                        new InputData('name', 'text', 'Your Name', '', 'PlayerName', ''),
+                                    ]}
+                                    url="http://localhost:60151/api/player"
```
This url should be extracted and built using string concatenation

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGameComponent.tsx:
```diff
@@ -0,0 +1,172 @@
+      }
+   }
+
+   //   public changeErrorStateOfInput = (id : string, errorMessage : string) => {
```
Please remove commented code.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGameComponent.tsx:
```diff
@@ -0,0 +1,172 @@
+         this.props.onSubmit(url, formData);
+      }
+
+      const validity = this.getValidityState(false);
```
variables/constant shall be declared on the top of a function

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGameComponent.tsx:
```diff
@@ -0,0 +1,172 @@
+      
+      const inputsData = this.getItemById(id, this.state.inputs);
+      if (this.state.inputs[inputsData.index].errorMessage !== errorMessage) {
+         //   console.log(this.state.inputs);
```
Please remove console logs, do not comment them

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGameComponent.tsx:
```diff
@@ -0,0 +1,172 @@
+
+
+   public changeHandler = (input: any, evt: any) => {
+
```
This method is quite long. It can be easily split in 2 or 3 smaller methods.

vasi-axinte commented on VikingQuiz.UI/src/Components/StartGame/PlayGameComponent.tsx:
```diff
@@ -0,0 +1,172 @@
+
+      const validity = this.getValidityState(false);
+      console.log(validity, this.state.isValid);
+      //   console.log(validity, this.state.isValid);
```
Please remove this.

vasi-axinte commented on VikingQuiz.UI/src/entities/PlayerCodeDto.tsx:
```diff
@@ -0,0 +1,14 @@
+    public code: string;
+
+    public constructor(pictureUrl: string, name: string, code: string){
+        this.id = 20;
```
What does the number 20 represents?

louisavasies committed:
Applied changes for code review.

louisavasies committed:
Fixed rebase errors.

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerGameRepo.cs:
```diff
@@ -0,0 +1,52 @@
+
+        public void Delete(int pid, int gid)
+        {
+            PlayerGame pg = new PlayerGame { PlayerId = pid, GameId = gid };
```
Please rename this pg to playerGame or something more relevant. Same with pid and gid.

vasi-axinte commented on VikingQuiz.Api/Repositories/PlayerGameRepo.cs:
```diff
@@ -0,0 +1,52 @@
+            return context.PlayerGame.ToList();
+        }
+
+        public PlayerGame GetPlayerGameByIds(int pid, int gid)
```
Same here with pid and gid.

louisavasies committed:
Changed some variable names in Repositories;
Corrected Front-End on StartGame&Connect pages;

vasi-axinte commented on VikingQuiz.Api/Controllers/PlayerGameController.cs:
```diff
@@ -14,11 +14,11 @@ namespace VikingQuiz.Api.Controllers
         private IEntityMapper<PlayerGameViewModel, PlayerGame> vmToEntityMapper;
 
-        public PlayerGameController(PlayerGameRepo playergameRepo, IEntityMapper<PlayerGame, PlayerGameViewModel> entityToVmMapper, IEntityMapper<PlayerGameViewModel, PlayerGame> vmToEntityMapper)
+        public PlayerGameController(PlayerGameRepository playergameRepo, IEntityMapper<PlayerGame, PlayerGameViewModel> entityToVmMapper, IEntityMapper<PlayerGameViewModel, PlayerGame> vmToEntityMapper)
```
Why didn`t you also renamed the variable if you renamed the class?

louisavasies committed:
Renamed variable playergameRepo

louisavasies committed:
Fixed validation rules
